hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/JCloudsDiscoveryStrategyTest.java

```diff
@@ -0,0 +1,100 @@
+package com.hazelcast.jclouds;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.nio.Address;
+import com.hazelcast.spi.discovery.DiscoveryNode;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.jclouds.compute.domain.NodeMetadata;
+import org.jclouds.compute.domain.internal.NodeMetadataImpl;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class JCloudsDiscoveryStrategyTest extends HazelcastTestSupport {
+    private static final Set<NodeMetadata> nodes = new HashSet<NodeMetadata>();
+    private static final Set<Address> addressesOfRunningInstances = new HashSet<Address>();
+
+    private static final int STARTING_PORT = 5701;
+    private static final int NUMBER_OF_NODES = 10;
+    private static final int NUMBER_OF_RUNNING_NODES = 7;
+
+    @BeforeClass
+    public static void start()
+            throws UnknownHostException {
+        for (int i = 0; i < NUMBER_OF_NODES; i++) {
+            HashSet<String> privateAdressses = new HashSet<String>();
+            privateAdressses.add("127.0.0." + (i+1));
+            NodeMetadata.Status status = NodeMetadata.Status.PENDING;
+            if (i < NUMBER_OF_RUNNING_NODES) {
+                status = NodeMetadata.Status.RUNNING;
+                addressesOfRunningInstances.add(new Address(privateAdressses.iterator().next(),0));
```

Comment:
This is how JClouds API works. `NodeMetadata` takes a list of IP addresses for both private and public IPs. It supports multiple IP addresses but this test uses only 1 public and private address.